server/keyshare/email.go

```diff
@@ -129,3 +135,18 @@ func sendHTMLEmail(addr string, a smtp.Auth, from, to *mail.Address, subject str
 		"\r\n")
 	return smtp.SendMail(addr, a, from.Address, []string{to.Address}, append(headers, msg...))
 }
+
+func verifyMXRecord(email string) error {
+	at := strings.LastIndex(email, "@")
+	if at > 0 {
+		records, err := net.LookupMX(email[at+1:])
+		if len(records) == 0 {
+			return errors.Errorf("No domain part found in %v", email)
+		} else if err != nil {
+			return err
+		}
+	} else {
+		return errors.Errorf("No '@'-sign found in %v", email)
+	}
+	return nil
+}
```

Comment:
In Go, you often see that the `else` clause is skipped if you use an early `return` in the body of the `if`. That saves indentations. So something like:

```
func verifyMXRecord(email string) error {
	at := strings.LastIndex(email, "@")
        if at < 0 {
            return errors.Errorf("no '@'-sign found in %v", email)
        }
        records, err := net.LookupMX(email[at+1:])
        if err != nil {
            return err
        }
	if len(records) == 0 {
		return errors.Errorf("no domain part found in %v", email)
	}
	return nil
}
```